File: readability.c
#include <cs50.h>
#include <ctype.h>
#include <math.h>
#include <stdio.h>
#include <string.h>

int main(void)
{
    // Prompt the user for some text
    string t = get_string("Text: ");

    // Count the number of letters, words, and sentences in the text

    int letters = 0;
    int words = 1;
    int sentence = 0;

    // Get length of string

    for (int i = 0, lnth = strlen(t); i < lnth; i++)

    // dont count spaces and commas as letters

    {
        if (isalpha(t[i]))
        {
            letters++;
        }

        else if (t[i] == ' ')
        {
            words++;
        }

        else if (t[i] == '.' || t[i] == '!' || t[i] == '?')
        {
            sentence++;
        }
    }
    // Compute the Coleman-Liau index

    float l = (float) letters / (float) words * 100;
    float s = (float) sentence / (float) words * 100;

    int index = round(0.0588 * l - 0.296 * s - 15.8);

    // Print Grade level
    if (index < 1)
    {
        printf("Before Grade 1\n");
    }

    else if (index > 16)
    {
        printf("Grade 16+\n");
    }

    else
    {
        printf("Grade %i\n", index);
    }
}

File: sentimental-readability/readability.py
from cs50 import get_string

# Prompt user for text
t = get_string("Text: ")

# Count letters, words, and sentences
letters = 0
words = 1
sentence = 0
i = 0

lnth = len(t)

# iderrate over the whole text
while i < lnth:
    if t[i].isalpha():  # check if letter is alphbetical if it is add 1 to the letter counter
        letters += 1
    elif t[i] == ' ':  # count words
        words += 1
    elif t[i] == '.' or t[i] == '!' or t[i] == '?':  # count sentences
        sentence += 1
    i += 1

l = letters / words * 100
s = sentence / words * 100

# Calculate Grade level
index = round(0.0588 * l - 0.296 * s - 15.8)

# Print out Grade level
if index < 1:
    print("Before Grade 1")
elif index > 16:
    print("Grade 16+")
else:
    print(f"Grade {index}")
